### PR TITLE
filter buddies only

### PR DESCRIPTION
**Pull request by poixen**
You may now filter games based on if they are for buddies only.

Addresses #1823

### CODE REVIEW

File: cockatrice/src/dlg_filter_games.cpp [hunk 1]
```diff
@@ -119,6 +123,16 @@ void DlgFilterGames::setUnavailableGamesVisible(bool _unavailableGamesVisible)
     unavailableGamesVisibleCheckBox->setChecked(_unavailableGamesVisible);
 }
 
+bool DlgFilterGames::getShowBuddiesOnlyGames() const
```
Comment (Daenyth) [inline cockatrice/src/dlg_filter_games.cpp]: Should this be const there? What does that do in that position?

Comment (poixen) [inline cockatrice/src/dlg_filter_games.cpp]: It means that this method is not allowed to change class state.

File: cockatrice/src/settings/gamefilterssettings.cpp [hunk 2]
```diff
@@ -15,6 +15,17 @@ QString GameFiltersSettings::hashGameType(const QString &gameType) const
     return QCryptographicHash::hash(gameType.toUtf8(), QCryptographicHash::Md5).toHex();
 }
 
+void GameFiltersSettings::setShowBuddiesOnlyGames(bool show)
+{
+    setValue(show, "show_buddies_only_games", "filer_games");
```
Comment (tooomm) [inline cockatrice/src/settings/gamefilterssettings.cpp]: typo? `filter_games`

**Comment by tooomm** [comment]
Looks like this:
buddy

The tick box is checked by default! I think that's ok since we want to display as much games as possible per default...

It works and successfully filters for the `buddy only` tag in games! :+1:

File: cockatrice/src/settings/gamefilterssettings.cpp (hunk 2 above)
Comment (poixen) [inline cockatrice/src/settings/gamefilterssettings.cpp]: crap that is an important typo, thanks

**Comment by poixen** [comment]
Just updated the formatting, all restriction related filters are now grouped.
screen shot 2016-07-03 at 20 14 42

File: cockatrice/src/dlg_filter_games.h [hunk 3]
```diff
@@ -4,6 +4,7 @@
 #include <QDialog>
 #include <QSet>
 #include <QMap>
+#include <QCheckBox>
```
Comment (ctrlaltca) [inline cockatrice/src/dlg_filter_games.h]: This line should be in the .cpp, not in the .h

**Comment by ctrlaltca** [comment]
Looks fine to me. Please squash before merge 👍

**Pull request merged**